src/sagemaker_xgboost_container/distributed_gpu/distributed_gpu_training.py

```diff
@@ -21,23 +21,56 @@
 from dask.distributed import Client
 
 from sagemaker_algorithm_toolkit import exceptions as exc
-from sagemaker_xgboost_container.constants.xgb_constants import MODEL_NAME
+from sagemaker_algorithm_toolkit.channel_validation import S3_DIST_TYPE, Channel
+from sagemaker_xgboost_container.algorithm_mode import train_utils
+from sagemaker_xgboost_container.callback import get_callbacks
+from sagemaker_xgboost_container.constants.xgb_constants import (
+    GPU_TREE_METHOD,
+    MODEL_NAME,
+    PIPE_MODE,
+)
+from sagemaker_xgboost_container.data_utils import CSV, PARQUET
 from sagemaker_xgboost_container.distributed_gpu.dask_cluster_utils import (
     get_host_ip,
     start_daemons_in_current_instance,
 )
 from sagemaker_xgboost_container.distributed_gpu.dask_data_utils import (
     create_dask_dmatrix,
     get_dataframe_dimensions,
-    load_data_into_memory,
+    read_data,
 )
 
 logger = logging.getLogger(__name__)
 
 SCHEDULER_PORT = "8786"
 WAIT_FOR_ALL_WORKERS_TIMEOUT_SEC = 20
 WORKER_STAY_ALIVE_CHECK_FREQ_SEC = 10
-
+SUPPORTED_TRAINING_CONTENT_TYPES = {CSV, PARQUET}
+
+NON_GPU_ERROR_MSG = "Dask training is only available for `gpu_hist` training on GPU instances."
+PIPE_MODE_ERROR_MSG = "Dask training is not supported for pipe mode input. Please use File mode."
+INPUT_FORMAT_ERROR_MSG = "Dask training is only supported for CSV and Parquet input."
+NOT_REPLICATED_ERROR_MSG = "Dask distributed training requires FullyReplicated data."
+
+
+def validate_gpu_train_configuration(
```

Comment:
I think it'd be better to still have this function raise a UserError because then we can catch that in the unit tests as well, but feel free to leave as is.